containers/docker: allow starting containers with more complex commands

brianneville opened the pull request:
naively splitting the command by whitespace prevents containers from being started with shell commands.
This change adds support for starting containers with more complex commands (e.g. `sh -c "command"`) or commands with quoted whitespace.
This change also adds test cases for the parsing being added

aajith-arista reviewed (commented):
This is definitely better than what we had before.
But ideally, shouldn't we do the quote based splitting even the command is not in the form of `<shell> -c` ?

Wouldn't it be better to use something like `shlex.split` ?
See https://pkg.go.dev/github.com/google/shlex
It could handle cases like multiple whitespaces, escaped quotes, quote within quotes etc

brianneville commented:
> This is definitely better than what we had before. But ideally, shouldn't we do the quote based splitting even the command is not in the form of `<shell> -c` ?
> 
> Wouldn't it be better to use something like `shlex.split` ? See https://pkg.go.dev/github.com/google/shlex It could handle cases like multiple whitespaces, escaped quotes, quote within quotes etc

great idea, thanks for pointing to that shlex package! 
I've implemented this in the latest patch - this simplifies the code nicely
